corehq/motech/repeaters/models.py

```diff
@@ -203,18 +203,19 @@ def save(self, *args, **kwargs):
     def __new__(cls, *args, **kwargs):
         repeater_class = cls
         try:
+            from corehq.motech.repeaters.utils import REPEATER_CLASS_MAP
+
             # get proxy name, either from kwargs or from args
             proxy_class_name = kwargs.get(cls.PROXY_FIELD_NAME)
             if proxy_class_name is None:
                 proxy_name_field_index = cls._meta.fields.index(
                     cls._meta.get_field(cls.PROXY_FIELD_NAME))
                 proxy_class_name = args[proxy_name_field_index]
-            # Since we are storing original repeater names so we need to append SQL
-            # This should be removed when SQL is dropped from model names
-            proxy_sql_class_name = f"SQL{proxy_class_name}"
-            # get proxy class, by name, from current module
-            repeater_class = getattr(sys.modules[__name__], proxy_sql_class_name)
-        except (IndexError, AttributeError):
+
+            repeater_class = REPEATER_CLASS_MAP.get(proxy_class_name)
+            if repeater_class is None:
+                raise UnknownRepeater(proxy_class_name)
+        except IndexError:
```

Comment:
In what situations is this error caught? Two suggestions:

- reduce the amount of code inside the `try` block to the minimum lines (ideally only one line) that may raise `IndexError` (I think it's `proxy_class_name = args[proxy_name_field_index]`) and move everything else into an `else:` block after `except IndexError`.
- I think this should only happen for `SQLRepeater` instantiation, is that right? If so, in this except block do
    ```py
    if repeater_class is not SQLRepeater:
        raise RepeaterError(f"Cannot find repeater type in arguments: {args} {kwargs}")
    ````
  If there are other known situations it can check for them too.

Reply:
>  I think this should only happen for SQLRepeater instantiation, is that right?

Actually, it will happen every time when we create a model object of any Repeater class which is not saved yet.  This is because `args would not contain `repeater_type` info in that.